arelle/plugin/validate/UK/ValidateUK.py

```diff
@@ -646,6 +651,31 @@ def _evaluateProfLossOrCharityFundsFact(self, code: str) -> CodeResult:
             )
         return CodeResult()
 
+    def _evaluateGroupFacts(self) -> CodeResult:
+        """
+        Logs an error if BalanceSheetDate with GroupCompanyDataDimension and Consolidated must exist as non-nil and
+        equal BalanceSheetDate with GroupCompanyDataDimension and Consolidated
+        """
```

Comment:
I'm not sure what this comment is trying to say, but it seems inconsistent with the code below (which doesn't check if values are nil).